createdisk.sh

```diff
@@ -47,6 +47,13 @@ ${SSH} core@${VM_IP} -- 'touch ~/.ssh/authorized_keys && chmod 0600 ~/.ssh/autho
 shutdown_vm ${CRC_VM_NAME}
 start_vm ${CRC_VM_NAME} ${VM_IP}
 
+# Remove the base deployment from rpm-ostree
+${SSH} core@${VM_IP} -- 'sudo rpm-ostree cleanup -r'
+# Shutdown and Start the VM after removing base deployment tree
+# This is required to get correct ostree commit in `/proc/cmdline` for macos
+shutdown_vm ${CRC_VM_NAME}
+start_vm ${CRC_VM_NAME} ${VM_IP}
```

Comment:
This is unexpected, `rpm-ostree cleanup -r` changes the commit currently being used? Since we just stopped/started the VM, I was thinking the commit would be the right one.

Reply:
> This is unexpected, `rpm-ostree cleanup -r` changes the commit currently being used?

@cfergeau no it just remove the rollback commit not currently being used and then update the bootloader. Since it update the bootversion that's why we need to restart to get the latest on for mac.
```
Bootloader updated; bootconfig swap: yes; bootversion: boot.1.1, deployment count change: -1
```

> Since we just stopped/started the VM, I was thinking the commit would be the right one.
Commit is correct one.

Reply:
The comment is misleading then, as it specifically mentions `ostree commit`: `This is required to get correct ostree commit in `/proc/cmdline` for macos`.
As you explained, it really is the kernel commandline which gets changed, namely `ostree=/ostree/boot.1/fedora-coreos/$hash/0` which switches between `boot.0` and `boot.1` when we run the cleanup.

Reply:
@cfergeau updated